data_steward/analytics/cdr_ops/rt_cdr_qc/cdr_deid_qa_report3_col_suppression.py

```diff
@@ -178,85 +155,62 @@
 FROM `{{project_id}}.{{deid_cdr}}.condition_occurrence`
 """)
-q = query.render(project_id=project_id, deid_cdr=deid_cdr)
-df1 = execute(client, q)
-if df1.loc[0].sum() == 0:
-    df = df.append({
-        'query': 'Query5 Condition provider_id',
-        'result': 'PASS'
-    },
-                   ignore_index=True)
+q = query.render(project_id=project_id,deid_cdr=deid_cdr)  
+df1=execute(client, q) 
+if df1.loc[0].sum()==0:
+ df = df.append({'query' : 'Query5 Condition provider_id', 'result' : 'PASS'},  
+                ignore_index = True) 
 else:
-    df = df.append(
-        {
-            'query': 'Query5 Condition provider_id',
-            'result': 'Failure'
-        },
-        ignore_index=True)
-df1
+ df = df.append({'query' : 'Query5 Condition provider_id', 'result' :  'Failure'},  
+                ignore_index = True) 
+df1        
 
-# # 6 Verify the following columns in the DEVICE_EXPOSURE table have been set to null:
+# # 6 Verify the provider_id in the DEVICE_EXPOSURE table is null and visit_detail_id is not all null:
 #
-# <font color='red'> Uncomment below when visit_detail_id is available
+# <font color='red'> DST-1423,updates, only provide_id is null.
 
 # +
 # query6 has error
 # no this column called visit_detail_id?
-
+    
 query = JINJA_ENV.from_string("""
 SELECT
 SUM(CASE WHEN provider_id IS NOT NULL THEN 1 ELSE 0 END) AS n_provider_id_not_null,
 SUM(CASE WHEN visit_detail_id IS NOT NULL THEN 1 ELSE 0 END) AS n_visit_detail_id_not_null
 
 FROM `{{project_id}}.{{deid_cdr}}.device_exposure`
 """)
-q = query.render(project_id=project_id, deid_cdr=deid_cdr)
-df1 = execute(client, q)
-
-if df1.loc[0].sum() == 0:
-    df = df.append({
-        'query': 'Query6 provider_id in Device',
-        'result': 'PASS'
-    },
-                   ignore_index=True)
+q = query.render(project_id=project_id,deid_cdr=deid_cdr)  
+df1=execute(client, q) 
+
+if df1.loc[0].sum()==0 or df1.iat[0,0] !=0:
+ df = df.append({'query' : 'Query6 provider_id in Device', 'result' : 'Failure'},  
+                ignore_index = True) 
 else:
-    df = df.append(
-        {
-            'query': 'Query6 provider_id in Device',
-            'result': 'Failure'
-        },
-        ignore_index=True)
+ df = df.append({'query' : 'Query6 provider_id in Device', 'result' :  'Pass'},  
+                ignore_index = True) 
 df1
 # -
 
-# # 7 Verify the following columns in the DRUG_EXPOSURE table have been set to null:
-# provider_id
+# # 7 Verify the provider_id in the DRUG_EXPOSURE table is null and visit_detail_id is not all null:
 #
-# <font color='red'> no visit_detail_id in old CDR?
-#
-#  Uncomment below when visit_detail_id is available in 2022 CDR
+# <font color='red'> DST-1423,updates, only provide_id is null.
 
 query = JINJA_ENV.from_string("""
 SELECT
 SUM(CASE WHEN provider_id IS NOT NULL THEN 1 ELSE 0 END) AS n_provider_id_not_null,
 SUM(CASE WHEN visit_detail_id IS NOT NULL THEN 1 ELSE 0 END) AS n_visit_detail_id_not_null
```

Comment:
This line was to be removed correct? Or is the field to be suppressed?

Reply:
which line? 235-237? if so, they were already removed in my updated one. should be like this, 

# # 6 Verify the provider_id in the DEVICE_EXPOSURE table is null and visit_detail_id is not all null:
#
# <font color='red'> DST-1423,updates, only provide_id is null.

# +
# query6 has error
# no this column called visit_detail_id?
    
query = JINJA_ENV.from_string("""
SELECT
SUM(CASE WHEN provider_id IS NOT NULL THEN 1 ELSE 0 END) AS n_provider_id_not_null,
SUM(CASE WHEN visit_detail_id IS NOT NULL THEN 1 ELSE 0 END) AS n_visit_detail_id_not_null

Reply:
here, my updated one, https://github.com/jqian2015/curation/blob/dst-1423/data_steward/analytics/cdr_ops/rt_cdr_qc/cdr_deid_qa_report3_col_suppression.py

Reply:
No line 242. I believe this ticket is to remove the part that checks for visit_detail_id to be NULL. We looked in the privacy rules and believe this field should be kept. 

Same for visit_detail_id in DRUG_EXPOSURE as well.

Reply:
ok, there might be misunderstanding here, as seen in the DST-1423, from Krishna, "So, here we are checking for whole column suppression for both the provide_id and visit_detail_id. Even though we have a lot of nulls is fine for visit_detail as that is from a lack of corresponding visit_detail records. The check should only fail if all the visit_detail_ids are null in the table." what i did was to check both columns and fail only provider_id is not null or if all the visit_detail_ids are null. But if we only need to check one column provider_id, we can do that too. But I mentioned that too and then Krishna responded that way. so could you please look through those conversions in the ticket and let me know the decision whether we should check one column or two columns?

Reply:
Thanks I see those conversations now. I misunderstood the ticket. Thanks!